benchmark/benchmark.hpp

```diff
@@ -83,6 +85,82 @@ struct alloc_data {
     size_t size;
 };
 
+struct next_alloc_data {
+    size_t offset;
+    size_t size;
+};
+
+#ifndef WIN32
+std::vector<cpu_set_t> affinityMask;
+
+int initAffinityMask() {
+    cpu_set_t mask;
+    CPU_ZERO(&mask);
+
+    if (sched_getaffinity(0, sizeof(mask), &mask) == -1) {
+        perror("sched_getaffinity");
+        return 1;
+    }
+
+    for (int cpu = 0; cpu < CPU_SETSIZE; cpu++) {
+        if (CPU_ISSET(cpu, &mask)) {
+            cpu_set_t mask;
+            CPU_ZERO(&mask);
+            CPU_SET(cpu, &mask);
+            affinityMask.push_back(mask);
+        }
+    }
+    return 0;
+}
+
+void setAffinity(benchmark::State &state) {
+    size_t tid = state.thread_index();
+    if (tid >= affinityMask.size()) {
+        state.SkipWithError("Not enough CPUs available to set affinity");
+    }
+
+    auto &mask = affinityMask[tid];
+
+    if (sched_setaffinity(0, sizeof(mask), &mask) != 0) {
+        state.SkipWithError("Failed to set affinity");
+    }
+}
+
+#else
+int initAffinityMask() {
+    printf(
+        "Affinity set not supported on Windows, benchmark can be unstable\n");
```

Comment:
There are `GetProcessAffinityMask` and `SetThreadAffinityMask` functions that you could use in a similar way as on Linux, I think.

Reply:
Yeah. We just do not care about windows ATM, so i was not waising time to implement it for windows. This is TODO if we will ever want to test performance on windows